### DISCUSSION TITLE
Salesforce Extensions

### DISCUSSION THREAD

**Discussion by RPDevJesco** [zed-industries/zed#15131] [Feature Requests]
Support for Salesforce Apex, LWC, Visualforce and Aura components would be a very nice addition.

**Comment by Nishitsuthar** [comment]
Yes you are correct even if they provide how do we can configure LSP that would also bring many devs towards this amazing editor.

**Comment by jsmithdev** [comment]
Yeah good call. After getting LSP square, we could make Tasks to handle functional needs -- get current path/file, run sf-cli command.
https://zed.dev/docs/tasks
https://zed.dev/docs/extensions/languages

**Comment by javiaparicio** [comment]
There is somebody already working on Apex support:

https://github.com/zed-industries/extensions/issues/1239

**Comment by Rihan9** [comment]
Is someone still working on this? I know at least 20 peaople that will love to discard VSC and migrate to zed if this extention was available

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Comment by brtheo** [comment]
Hello all,
I've been working for the past few days on a TUI app to handle SF cli Interactions that you can't achieve through zed tasks only like : metadata generation, org switching and manifest generator
You can take a look at it [sf-tui](https://GitHub.com/brtheo/sf-tui)

In addition, here's how I set up my zed tasks 
[tasks](https://github.com/brtheo/zed-sf-cli-tasks)
